compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala

```diff
@@ -154,4 +159,25 @@ object PickledQuotes {
     }
     Block(x1 :: Nil, rec(f))
   }
+
+  private def classToType(clazz: Class[_])(implicit ctx: Context): Type = {
+    if (clazz == classOf[Boolean]) defn.BooleanType
+    else if (clazz == classOf[Byte]) defn.ByteType
+    else if (clazz == classOf[Char]) defn.CharType
+    else if (clazz == classOf[Short]) defn.ShortType
+    else if (clazz == classOf[Int]) defn.IntType
+    else if (clazz == classOf[Long]) defn.LongType
+    else if (clazz == classOf[Float]) defn.FloatType
+    else if (clazz == classOf[Double]) defn.DoubleType
+    else if (clazz == classOf[Unit]) defn.UnitType
+    else if (!clazz.isMemberClass) ctx.getClassIfDefined(clazz.getCanonicalName).typeRef
```

Comment:
`Array`-s probably need special handling, too.

Reply:
Will look into it

Reply:
Yes, they needed it. Thanks :)